app/assets/javascripts/lat-long-info.js

```diff
@@ -0,0 +1,52 @@
+/* global FileReader EXIF $ */
+
+$(function () {
+  var sightingImageEl = document.getElementById('sighting_image')
+  if (sightingImageEl) {
+    sightingImageEl.addEventListener('change',
+    function (event) {
+      var fileInput = event.target
+      var imageFile = fileInput.files[0]
+
+      var fr = new FileReader()
+      fr.addEventListener('load', function () {
+        var thumbnailContainer = document.getElementById('img-thumb')
+        var thumbnail = document.createElement('img')
+        thumbnail.src = fr.result
+
+        thumbnailContainer.appendChild(thumbnail)
+
+        getExif(thumbnail, function (longitude, latitude) {
+          $('[name*=lat]').val(latitude)
+          $('[name*=long]').val(longitude)
+        })
+      })
+      fr.readAsDataURL(imageFile)
+    })
+  }
+})
+
+function getExif (img, cb) {
+  // if cb is not a function, log an error
+  if (typeof (cb) === 'function') {
+    EXIF.getData(img, function () {
+      var longitude = EXIF.getTag(this, 'GPSLongitude')
+      var latitude = EXIF.getTag(this, 'GPSLatitude')
+      // var latAndLong = document.getElementById('latAndLong')
+      // latAndLong.innerHTML = `${latitude},  ${longitude}`
+      var decLat = degToDec(latitude)
+      var decLong = degToDec(longitude)
+      cb(decLat, decLong)
+    })
+  } else {
+    window.alert('there is no function')
```

Comment:
Instead of doing a `if ... else` this way. Create a guard clause. That is put the if at the top of the function and `throw` and error if it fails. As it turns out (and you may have experienced this yourself), `alert`s are the worst. They're really annoying. Instead do:

```javascript
if (typeof(cb) === 'function')
  throw new TypeError('Expected second argument to be a function, but got ' + typeof(cb))
end
```

This will show an error in the console and stop the rest of the code in the function from ever running